diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LongNumericalValuesSeparatedRule.kt

```diff
@@ -0,0 +1,152 @@
+    private var isFixMode: Boolean = false
+
+    companion object {
+        private const val DELIMITER_LENGTH: Int = 3
+        private const val MAX_NUMBER_LENGTH: Int = 3
+    }
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = LongNumericalValuesConfiguration(
+                configRules.getRuleConfig(Warnings.LONG_NUMERICAL_VALUES_SEPARATED)?.configuration ?: mapOf())
+
+        if (node.elementType == INTEGER_LITERAL) {
+            if(!isValidConstant(node.text, configuration, node)) {
+                Warnings.LONG_NUMERICAL_VALUES_SEPARATED.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    fixIntegerConstant(node, configuration.maxBlockLength)
+                }
+            }
+        }
+
+        if (node.elementType == FLOAT_LITERAL) {
+            if(!isValidConstant(node.text, configuration, node)) {
+                val parts = node.text.split(".")
+
+                Warnings.LONG_NUMERICAL_VALUES_SEPARATED.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    fixFloatConstantPart(parts[0], parts[1], configuration, node)
+                }
+
+            }
+        }
+
+    }
+
+
+    private fun fixIntegerConstant(node: ASTNode, maxBlockLength: Int) {
+        val resultRealPart = StringBuilder(nodePrefix(node.text))
+
+        val chunks = removePrefixSuffix(node.text).reversed().chunked(maxBlockLength).reversed()
+        resultRealPart.append(chunks.joinToString(separator = "_") { it.reversed() })
+
+        resultRealPart.append(nodeSuffix(node.text))
+        (node as LeafPsiElement).replaceWithText(resultRealPart.toString())
+    }
+
+    private fun fixFloatConstantPart(realPart: String, fractionalPart: String, configuration: LongNumericalValuesConfiguration, node: ASTNode) {
+        val resultRealPart = StringBuilder(nodePrefix(realPart))
+        val resultFractionalPart = StringBuilder()
+
+        if (removePrefixSuffix(realPart).length > configuration.maxLength) {
+            val chunks = removePrefixSuffix(realPart).reversed().chunked(configuration.maxBlockLength).reversed()
+            resultRealPart.append(chunks.joinToString(separator = "_") { it.reversed() })
+
+            resultRealPart.append(nodeSuffix(realPart)).append(".")
+        } else {
+            resultRealPart.append(removePrefixSuffix(realPart)).append(".")
+        }
+
+        if ( removePrefixSuffix(fractionalPart).length > configuration.maxLength) {
+            val chunks = removePrefixSuffix(fractionalPart).reversed().chunked(configuration.maxBlockLength).reversed()
```

Comment:
Actually, I don't think you need to reverse here:
`1.111_11` looks better than `1.11_111`, I think, because it groups digits into more meaningful chunks.